crypto/Cargo.toml

```diff
@@ -15,13 +15,14 @@ hex = "0.4.3"
 rand = { version = "0.7.3", features = ["std"] }
 serde = { version = "1.0.139", features = ["derive"] }
 zeroize = "1.5.6"
-signature = "1.5.0"
```

Comment:
in order to implement this function `fn generate<R: CryptoRng + RngCore>(rng: &mut R) -> Self;
` in the trait KeyPair, all implementations of it (ed25519 and secp256k1) needs to be on the same RngCore version. due to ed255-dalek being capped at RngCore 0.5.1, we need to use a version of k256 that also depends on rand-core 0.5.1. this lower version of k256 resulted in a lower version of `signature` it depends on. 

hopefully, once we get rid of dalek, this will be an non-issue for us to use the latest rand-core

Reply:
Please open the issue: we use the `signature` trait is used in `Sui` as well!
Note also that some crates re-export the rand crate : is this the case for `secp256k1`? You could leave this and pass in a `secp256k1::rand::rng::OsRng` or some such?